### ISSUE TITLE
Yolo model predict function not getting executed

### ISSUE THREAD

**Issue by sanket-tss** [ultralytics/ultralytics#14674]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi 
I am running a flask api application with threading disabled and multiple processes, this application has an api endpoint which triggers an YOLOv8 model and does some processing based on the model prediction. This whole application is wrapped inside a docker and hosted on kubernetes 
Issue is that the application when it receives call on the endpoint, it get stuck at `model.predict` line forever, but the same docker images run perfectly fine on kubernetes.

docker file base image is `python:3.10-slim`
flask requirements `Flask==2.0.2 werkzeug==2.0.2`
kubernetes version: 1.28.9 [nodes are of ubuntu]
Node image version AKSUbuntu-2204gen2containerd-202407.08.0

This is started happening with the application since this monday (22/07), also have tried running the same docker which is running on kubernetes or different machine, building the image on different machines (as well), windows, linux(different type of linux), have tried different base image as well.
I currently cannot share the code, i will try to isolate the whole issue and then share the code which is not working..

**Comment by sanket-tss** [comment]
this is breaking at 8.2.61 version, it is working at 8.2.60

code diff at version
https://github.com/ultralytics/ultralytics/compare/v8.2.60...v8.2.61

**Comment by ambitious-octopus** [comment]
Hey @sanket-tss, using this script and the latest version, everything seems to work fine!

```python
from flask import Flask
from ultralytics import YOLO

model = YOLO('yolov8n.pt')
names = model.names
app = Flask(__name__)


@app.route("/", methods=['GET'])
def run_model():
    result = model.predict('https://ultralytics.com/images/zidane.jpg', device='cpu')
    return {"objects": [names[int(i)] for i in result[0].boxes.cls.tolist()]}

  
app.run(port=8000, processes=3, threaded=False)
```


This is my yolo check output:
```
Ultralytics YOLOv8.2.64 🚀 Python-3.10.0 torch-2.2.2+cu121 CPU (12th Gen Intel Core(TM) i7-12650H)
Setup complete ✅ (16 CPUs, 15.3 GB RAM, 174.5/383.3 GB disk)

OS                  Linux-6.9.3-76060903-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.0
Install             git
RAM                 15.29 GB
CPU                 12th Gen Intel Core(TM) i7-12650H
CUDA                None

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.8.4>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.13.0>=1.4.1
torch               ✅ 2.2.2>=1.8.0
torchvision         ✅ 0.17.2>=0.9.0
tqdm                ✅ 4.66.2>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0
```

**Comment by sanket-tss** [comment]
we found a fix, we had to add  `model = YOLO('yolov8n.pt').to("cpu")`
only then the whole execution works..
earlier at 8.2.60 it is working perfectly fine if we don't provide .to anything

**Comment by pderrenger** [comment]
Thank you for the update. It seems like specifying the device explicitly with `.to("cpu")` resolved the issue in version 8.2.61. This change might be due to updates in the model's handling of device allocation. If you encounter any further issues, please ensure you're using the latest package versions. If the problem persists, feel free to reach out with more details.
